code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,276 @@
+/mob/living/simple_animal/hostile/eldritch
+	name = "Demon"
+	real_name = "Demon"
+	desc = ""
+	gender = NEUTER
+	mob_biotypes = NONE
+	speak_emote = list("screams")
+	response_help_continuous = "thinks better of touching"
+	response_help_simple = "think better of touching"
+	response_disarm_continuous = "flails at"
+	response_disarm_simple = "flail at"
+	response_harm_continuous = "reaps"
+	response_harm_simple = "tears"
+	speak_chance = 1
+	icon = 'icons/mob/eldritch_mobs.dmi'
+	speed = 0
+	a_intent = INTENT_HARM
+	stop_automated_movement = 1
+	AIStatus = AI_ON
+	attack_sound = 'sound/weapons/punch1.ogg'
+	see_in_dark = 7
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 0)
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = INFINITY
+	healable = 0
+	faction = list("e_cult")
+	movement_type = GROUND
+	pressure_resistance = 100
+	del_on_death = TRUE
+	deathmessage = "implodes into itself"
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet
+	name = "Raw Prophet"
+	real_name = "Raw Prophet"
+	desc = "Abomination made from severed limbs."
+	icon_state = "raw_prophet"
+	status_flags = CANPUSH
+	icon_living = "raw_prophet"
+	melee_damage_lower = 5
+	melee_damage_upper = 10
+	maxHealth = 75
+	health = 75
+	sight = SEE_MOBS|SEE_OBJS|SEE_TURFS
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Initialize()
+	. = ..()
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash/long)
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/telepathy/eldritch)
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Login()
+	. = ..()
+	client.change_view(11)
+
+/mob/living/simple_animal/hostile/eldritch/armsy
+	name = "Terror of the night"
+	real_name = "Armsy"
+	desc = "Abomination made from severed limbs."
+	icon_state = "armsy_start"
+	icon_living = "armsy_start"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	var/mob/living/simple_animal/hostile/eldritch/armsy/back
+	var/mob/living/simple_animal/hostile/eldritch/armsy/front
+	var/oldloc
+	var/allow_pulling = FALSE
+	var/stacks_to_grow = 5
+	var/current_stacks = 0
+
+/mob/living/simple_animal/hostile/eldritch/armsy/New(spawn_more = TRUE,len = 6)
+	. = ..()
+	oldloc = loc
+	if(!spawn_more)
+		return
+	allow_pulling = TRUE
+	var/mob/living/simple_animal/hostile/eldritch/armsy/next
+	var/mob/living/simple_animal/hostile/eldritch/armsy/prev
+	var/mob/living/simple_animal/hostile/eldritch/armsy/current
+	for(var/i = 0, i <= len,i++)
```

Comment:
List loop.

Reply:
Add an error check in case `len` is lower than 2, where you'll have quirkiness.
A more informative variable name would be positive, something like `parts_length` or whatever.

Reply:
i just added a min() so that len cannot go below 3